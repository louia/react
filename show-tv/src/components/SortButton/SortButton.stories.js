import React from 'react';
import SortButton from './index';
// import './index.css'

export default {
    title: 'components/SortButton',
    component: SortButton,
}

export const default_value = () => <SortButton disabled={false} actif={true}>Tri default</SortButton>;
export const button_ascending = () => <SortButton disabled={false} order={"ascending"} >Tri activé</SortButton>;
export const button_descending_enable = () => <SortButton disabled={false} order={"descending"} >Tri activé</SortButton>;
export const button_descending_disabled = () => <SortButton disabled={true} order={"descending"}>Tri activé</SortButton>;
export const button_ascending_disabled = () => <SortButton disabled={true} order={"ascending"} >Tri désactivé</SortButton>;
