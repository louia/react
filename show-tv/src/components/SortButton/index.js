import React from 'react';
import PropTypes from 'prop-types';

/** Bouton permettant de trier */
export class SortButton extends React.Component {

    render() {
        let {children, order, onClick, actif, className, ...props } = this.props;

        let fontIcon = order === 'ascending' ? '▲' : '▼';
        let css = 'sortButton' + (actif ? ' actif' : '');

        return (
            <div className={css}>
                <button actif={actif} {...props} onClick={onclick}>{children}</button>
                <button actif={actif} {...props} onClick={onclick}>{fontIcon}</button>
            </div>
        );
    }
}

SortButton.propTypes = {
    /** Boolean qui indique si le bouton est actif ou non */
    actif: PropTypes.bool,
    /** String qui indique l'ordre croissant ou décroissant */
    order: PropTypes.string,
    /** Boolean qui indique si le bouton est désactivé ou non */
    disabled: PropTypes.bool,
    /** Fonction de rappel au click */
    onClick: PropTypes.func,
};


export default SortButton;
