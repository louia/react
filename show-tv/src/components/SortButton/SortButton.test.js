import { render, fireEvent } from '@testing-library/react';
 import { default_value, initialized } from './SortButton.stories';
 import initStoryshots from '@storybook/addon-storyshots';

 initStoryshots();
describe('Button test', () => {
    test('should increment when clicked', () => {
        expect(true).toBeTruthy();
    });

});
