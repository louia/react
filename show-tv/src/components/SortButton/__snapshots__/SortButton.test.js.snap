// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components/Button Button Actif 1`] = `
<button
  className="undefined monButton actif"
>
  Button activé
</button>
`;

exports[`Storyshots components/Button Button Non Actif 1`] = `
<button
  className="test monButton"
>
  Button désactivé
</button>
`;

exports[`Storyshots components/Button Default Value 1`] = `
<button
  className="undefined monButton"
>
  Mon boutton
</button>
`;

exports[`Storyshots components/Button Test With H 1 1`] = `
<button
  className="undefined monButton"
>
  <div>
    <h1>
      Coucou
    </h1>
  </div>
</button>
`;

exports[`Storyshots components/Header DOM 1`] = `
<header
  className="mini"
>
  <div
    className="item-header"
  >
    <p>
      Bienvenue
    </p>
  </div>
  <div
    className="item-header"
  >
    <p>
      Sous titre
    </p>
  </div>
</header>
`;

exports[`Storyshots components/Header Default Value 1`] = `
<header>
  <div
    className="item-header"
  >
    Bienvenue
  </div>
  <div
    className="item-header"
  >
    Sous titre
  </div>
</header>
`;

exports[`Storyshots components/Header Initialized 1`] = `
<header>
  <div
    className="item-header"
  >
    Bienvenue
  </div>
  <div
    className="item-header"
  >
    Sous titre
  </div>
</header>
`;

exports[`Storyshots components/Header MINI 1`] = `
<header
  className="mini"
>
  <div
    className="item-header"
  >
    Bienvenue
  </div>
  <div
    className="item-header"
  >
    Sous titre
  </div>
</header>
`;

exports[`Storyshots components/SortButton Button Ascending 1`] = `
<div
  className="sortButton"
>
  <button
    disabled={false}
    onClick={null}
  >
    Tri activé
  </button>
  <button
    disabled={false}
    onClick={null}
  >
    ▲
  </button>
</div>
`;

exports[`Storyshots components/SortButton Button Ascending Disabled 1`] = `
<div
  className="sortButton"
>
  <button
    disabled={true}
    onClick={null}
  >
    Tri désactivé
  </button>
  <button
    disabled={true}
    onClick={null}
  >
    ▲
  </button>
</div>
`;

exports[`Storyshots components/SortButton Button Descending Disabled 1`] = `
<div
  className="sortButton"
>
  <button
    disabled={true}
    onClick={null}
  >
    Tri activé
  </button>
  <button
    disabled={true}
    onClick={null}
  >
    ▼
  </button>
</div>
`;

exports[`Storyshots components/SortButton Button Descending Enable 1`] = `
<div
  className="sortButton"
>
  <button
    disabled={false}
    onClick={null}
  >
    Tri activé
  </button>
  <button
    disabled={false}
    onClick={null}
  >
    ▼
  </button>
</div>
`;

exports[`Storyshots components/SortButton Default Value 1`] = `
<div
  className="sortButton actif"
>
  <button
    actif={true}
    disabled={false}
    onClick={null}
  >
    Tri default
  </button>
  <button
    actif={true}
    disabled={false}
    onClick={null}
  >
    ▼
  </button>
</div>
`;
